### PR TITLE
add #:immutable option to define-type

### PR DESCRIPTION
**Pull request by AlexKnauth**

### CODE REVIEW

**Comment by jeapostrophe** [comment]
Can you update the documentation too?

**Comment by AlexKnauth** [comment]
Ok, I updated the documentation.

File: scribblings/plai.scrbl [hunk 1]
```diff
@@ -74,6 +75,9 @@ its variant.
 The value of each field is checked by its associated @racket[_contract-expr].
 A @racket[_contract-expr] may be an arbitrary predicate or a contract.
 
+If the @racket[#:immutable] option is provided, the contstructors create
```
Comment (jeapostrophe) [inline scribblings/plai.scrbl]: Small typo here

File: scribblings/plai.scrbl [hunk 2]
```diff
@@ -74,6 +75,9 @@ its variant.
...
+immutable structs. Otherwise, they can be mutated.
+
 In addition to the contructors, a @racket[define-type] expression also defines:
```
Comment (jeapostrophe) [inline scribblings/plai.scrbl]: And a typo from me :)

**Comment by AlexKnauth** [comment]
Ok fixed.

**Comment by jeapostrophe** [comment]
Awesome! Thanks so much.

**Pull request merged**